openedx/openedx-platform
# feat: add content groups support to teams

## Description

This PR creates a bridge between teams and content groups, based on the exploration of new flexible grouping definition published previously on this PR: https://github.com/openedx/edx-platform/pull/32806

We're calling `flexible group` to a course user group similar to cohorts but without unicity restrictions, so it's more generic.

Here are the use cases we've implementing so far:
- When creating a new team within a topic, a new flexible group is created matching the team's name
- When creating/editing a team, a content group can be associated with it. What happens behind the scenes is the flexible group matching the team is associated with the selected content group
- When a student joins a team, they automatically join the flexible group matching the team
- When configuring the content groups, the student would be able to see only the subsections associated with the content group matching the team <-> flexible group

## Supporting information

This implementation is part of the effort for: https://github.com/openedx/platform-roadmap/issues/250, and it's a continuation of https://github.com/openedx/edx-platform/pull/32806 connecting flexible groups capabilities with teams.

More on teams: https://openedx.atlassian.net/wiki/spaces/LEARNER/pages/19006021/Teams

## Testing instructions

1. Move to this edx-platform's branch `MJG/extended-teams`
2. Since we're adding a new extension point, we need to re-install openedx so it's loaded: 
```
tutor dev exec lms|cms bash
pip install .
```
3. Restart the openedx services
4. Create a new topic for your course: `Studio > Select your course > Pages & Resources > Teams`
5. Now, create a new content group for the flexible groups partitions: `Studio> Select your course > Group Configurations > Content Group for group`
6. Associate subsections to your content group
7. Create a new team from the LMS: `Select your course > Teams tab > Select the topic you created > Create a new team in this topic`
8. In that view, select the content group you created before 
9. Now, as a student, enroll into the course and then join the team we created
10. Go to the course, you'll be able to see just the subsections belonging to the content group associated with the team
